plugins/synaptics-rmi/fu-synaptics-rmi-device.c

```diff
@@ -873,6 +873,52 @@ fu_synaptics_rmi_device_write_firmware (FuDevice *device,
 	/* erase all */
 	//FIXME: write $3 into F34_Flash_Data3
 	//FIXME: wait for ATTN and check success
+	/* v7+ */
+	if (priv->f34->function_version == 0x02) {
+		g_autoptr(GByteArray) erase_cmd = g_byte_array_new ();
+		fu_byte_array_append_uint8 (erase_cmd, CORE_CODE_PARTITION);
+		fu_byte_array_append_uint32 (erase_cmd, 0x0, G_LITTLE_ENDIAN);
+		if (priv->bootloader_id[1] == 8){
+			// For bootloader v8
+			fu_byte_array_append_uint8 (erase_cmd, CMD_V7_ERASE_AP);
+		} else {
+			// For bootloader v7
+			fu_byte_array_append_uint8 (erase_cmd, CMD_V7_ERASE);
+		} 
+		fu_byte_array_append_uint8 (erase_cmd, priv->bootloader_id[0]);
+		fu_byte_array_append_uint8 (erase_cmd, priv->bootloader_id[1]);
+		//FIXME: rmi4update_poll()
+		//FIXME: Check whether device is in bootloader mode (m_inBLmode)
```

Comment:
Is this really going to be needed?  That's what `FWUPD_DEVICE_FLAG_IS_BOOTLOADER` is for, no?  This function shouldn't have been entered if not in bootloader.